### PR TITLE
[Notification] Migrate alert create email

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic: https://github.com/metabase/metabase/issues/47438

### CODE REVIEW

File: src/metabase/events/notification.clj [hunk 1]
```diff
@@ -78,14 +79,19 @@
     {:user-invited-today         (t/format "MMM'&nbsp;'dd,'&nbsp;'yyyy" (t/zoned-date-time))
      :user-invited-email-subject (trs "You''re invited to join {0}''s {1}" (public-settings/site-name) (messages/app-name-trs))
      :user-invited-join-url      (join-url (:object event-info))}
-    {}))
+
+    :event/alert-create
+    {:alert-create-condition-description (->> event-info :object
```
Comment (piranha) [inline src/metabase/events/notification.clj:84]: ehhh I understand why the name, but don't like it, will look weird in templates (already does)...

File: resources/migrations/001_update_migrations.yaml [hunk 2]
```diff
@@ -9733,6 +9733,20 @@ databaseChangeLog:
...
+      comment: Migrate user invited email to system event notification
+      rollback: # not necessary, will be removed with the table
+      preConditions:
+        - onFail: MARK_RAN
+        - sqlCheck:
+            expectedResult: 0
+            sql: SELECT count(*) FROM notification_subscription WHERE type = 'notification-subscription/system-event' AND event_name = 'event/alert-create'
+      changes:
+        - customChange:
+            class: "metabase.db.custom_migrations.CreateSystemNotificationAlertCreated"
```
Comment (piranha) [inline resources/migrations/001_update_migrations.yaml:9748]: To be honest, adding every notification as a migration doesn't feel very appealing to me. Let's discuss our options off-github somewhere?

File: src/metabase/events/notification.clj (hunk 1 above)
Comment (qnkhuat) [reply src/metabase/events/notification.clj:84]: yea, it's kind of my intention to make it extra explicit :D, to let people it's wonkiness that we did it this way you know.

anw, what's weird about it?

File: resources/migrations/001_update_migrations.yaml (hunk 2 above)
Comment (qnkhuat) [reply resources/migrations/001_update_migrations.yaml:9748]: sure

**Comment by qnkhuat** [comment]
Let's discuss it [here](https://metaboat.slack.com/archives/CKZEMT1MJ/p1728283200544159). wdyt we handle this all-together on a feature branch where I migrate 3 emails? let's this PR about migration this email only.

**Review by sansolovyov — APPROVED** [review]
Nice!

The stuff I'm commented on should be dealt (or not) separately, I think.

**Pull request merged**